sql-api/src/main/java/io/cdap/pipeline/sql/api/template/SQLSink.java

```diff
@@ -26,5 +26,5 @@
    * Gets the table to create or insert into. Type schema of the table is computed from the nodes.
    * @return The table to create or insert into
    */
-  public abstract TableInfo getDestinationTable();
+  public abstract TableDelegate getDestinationTable();
```

Comment:
the object returned should only define the contract, not the implementation. The TableInfo abstract class should still be used here, with the TableDelegate as a concrete implementation that plugins can use.

Reply:
Fixed